presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1537,25 +1545,35 @@ private void undoUpdateStatisticsOperations()
 
         private void executeIrreversibleMetastoreOperations()
         {
-            List<String> failedIrreversibleOperationDescriptions = new ArrayList<>();
-            List<Throwable> suppressedExceptions = new ArrayList<>();
-            boolean anySucceeded = false;
+            List<String> failedIrreversibleOperationDescriptions = Collections.synchronizedList(new ArrayList<>());
+            List<Throwable> suppressedExceptions = Collections.synchronizedList(new ArrayList<>());
+            AtomicBoolean anySucceeded = new AtomicBoolean(false);
```

Comment:
nit: you could make it `volatile` instead

Reply:
volatile is not allowed in method scope.